models/articles.model.js

```diff
@@ -35,4 +35,21 @@ const fetchArticles = () => {
   });
 };
 
-module.exports = { fetchArticleById, fetchArticles };
\ No newline at end of file
+const patchVoteInArticleById = (article) => {
+  return fetchArticleById(article.article_id).then(() => {
+
+    if (!Number.isInteger(article.inc_votes)) {
+      throw new BadRequestError();
+    }
+
+    const sql = `UPDATE articles SET votes = votes + $1 WHERE article_id = $2 RETURNING *`;
+
+    return db.query(sql, [article.inc_votes, article.article_id]).catch(() => {
```

Comment:
having catching in the model, makes things a little less manageable as we scale. Using the middleware pattern and having a single catch in the controller makes managing the flow of errors